### PR TITLE
email-notification for sage intacct api changes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: apps/sage_intacct/tasks.py [hunk 1]
```diff
@@ -205,6 +205,48 @@ def schedule_journal_entries_creation(workspace_id: int, expense_group_ids: List
         if chain.length() > 1:
             chain.run()
 
+def schedule_credit_card_charge_creation(workspace_id: int, expense_group_ids: List[str]):
```
Comment (NileshPant1999) [inline apps/sage_intacct/tasks.py:208]: @DhaaraniCIT why is this function added ??

File: apps/sage_intacct/tasks.py [hunk 2]
```diff
@@ -1272,3 +1314,38 @@ def schedule_fyle_reimbursements_sync(sync_sage_intacct_to_fyle_payments, worksp
 
         if schedule:
             schedule.delete()
+
+def schedule_journal_entry_creation(workspace_id: int, expense_group_ids: List[str]):
+    """
```
Comment (NileshPant1999) [inline apps/sage_intacct/tasks.py:1319]: why is this added here ??

File: apps/workspaces/models.py [hunk 3]
```diff
@@ -128,6 +130,9 @@ class WorkspaceSchedule(models.Model):
     enabled = models.BooleanField(default=False)
     start_datetime = models.DateTimeField(help_text='Datetime for start of schedule', null=True)
     interval_hours = models.IntegerField(null=True)
+    error_count = models.IntegerField(null=True, help_text='Number of errors in export')
+    additional_email_options = JSONField(default=list, help_text='Email and Name of person to send email', null=True)
+    emails_selected = ArrayField(base_field=models.CharField(max_length=255), null=True, help_text='File IDs')
```
Comment (NileshPant1999) [inline apps/workspaces/models.py:135]: change the help text

File: apps/workspaces/tasks.py [hunk 4]
```diff
@@ -1,24 +1,59 @@
-from datetime import datetime
+import email
```
Comment (NileshPant1999) [inline apps/workspaces/tasks.py:1]: remove unwanted imports

File: apps/workspaces/templates/mail_template.html [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+                </tr>
+                <tr align="center">
+                  <td>
+                    <table style="width: 250px" border="0" cellspacing="0" cellpadding="0">
+                      <tr align="center">
+                        <td align="center" style="border-radius: 4px;" bgcolor="#ff3366">
+                          <a href="{{app_url}}
+                            target="_blank"
+                            style="font-size: 12px; color: #ffffff; text-decoration: none; text-decoration: none;border-radius: 4px; padding: 12px 18px; border: 1px solid #ff3366; display: inline-block;">
+                            <b>Visit Integation</b>
```
Comment (ashwin1111) [inline apps/workspaces/templates/mail_template.html:104]: `Integration`

File: apps/workspaces/templates/mail_template.html [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            </td>
+          </tr>
+          <tr>
+            <td
+              background="https://raw.githubusercontent.com/fylein/fyle-netsuite-api/b1ebe6c1db08c4e85cc34244f8bf5a1b3f1219ea/apps/workspaces/templates/images/background.png"
+              style="padding: 10px 10px 8px 5px; background-color: #f3f5f6;">
+              <table align="center">
+                <tr style="color: #4A4A4A; font-size: 12px; text-align: center; line-height: 20px;">
+                  <td>
+                    2022 Fyle. All Rights Reserved.
```
Comment (ashwin1111) [inline apps/workspaces/templates/mail_template.html:168]: use js function for year, don't hard code

File: apps/workspaces/tasks.py [hunk 7]
```diff
@@ -101,3 +134,53 @@ def run_sync_schedule(workspace_id):
...
                 )
+
+def run_email_notification(workspace_id):
+
+    ws_schedule, _ = WorkspaceSchedule.objects.get_or_create(
+        workspace_id=workspace_id
+    )
+
+    task_logs = TaskLog.objects.filter(
+        ~Q(type__in=['CREATING_BILLS', 'FETCHING_EXPENSES']),
```
Comment (ashwin1111) [inline apps/workspaces/tasks.py:145]: CREATING_BILLS isn't right. We have 2 different types in payments sync. You'll need to add them here.

Comment (NileshPant1999) [reply apps/workspaces/tasks.py:145]: @DhaaraniCIT you check latest netsuite code

**Comment by NileshPant1999** [comment]
Before deploying remember to add environment variable to fyle deploy, I will explain on call later.

**Review by ashwin1111 — APPROVED** [review]
